shopinvader/services/customer.py

```diff
@@ -35,6 +38,24 @@ def create(self, **params):
         self._post_create(self.work.partner)
         return self._prepare_create_response(binding)
 
+    def update(self, _id, **params):
+        # We get the binding from the partner
+        if not self.partner:
+            raise UserError(_("You must provide a partner"))
+
+        partner = self._get(_id)
+        if not partner.shopinvader_bind_ids:
+            raise UserError(_("Partner %s is not a customer") % partner.id)
+        binding = partner.shopinvader_bind_ids[0]
+
+        vals = self._prepare_params(params, mode="update")
+        binding.write(vals)
+        self._load_partner_work_context(binding)
+        self._post_update(self.work.partner)
+
+        # Should we rename this function?
+        return self._prepare_create_response(binding)
```

Comment:
you should not use this method. You don't have to assign a cart on update.
When you update a partner you should get the full data IMO. 
```suggestion
        return self._to_json(partner)
```